### PR TITLE
INTDEV-468 Make all resource folders optional

### PR DESCRIPTION
**Pull request by Fuzzbender**
Make all resource folders optional in the project.

Additionally, make file-based resources work use same implementation in `create.ts`. Remove resource-specific "exists" functions and provide one generic function. 
Additionally, split the "project interfaces" to two: project related interfaces and resource related interfaces. The latter will be moved to `Resource` class once we get to implement it. 
Finally, renamed the resource interface to be more or less similar; `Template` is an exception for now, as there is `Template` class that is exported.

### CODE REVIEW

**Comment by Fuzzbender** [comment]
There's at least one issue with the implementation; if you add card to a template, it logs: "Cannot find folder '/Users/samimerila/projects/test/.cards/local/calculations'". Need to investigate where it is coming from.

File: tools/data-handler/src/interfaces/resource-interfaces.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+ * in a correct place in folder structure.
+ */
+
+// Calculation does not have own metadata.
+export type CalculationMetadata = ResourceBaseMetadata;
+
+// Card type content.
+export interface CardType extends ResourceBaseMetadata {
+  workflow: string;
+  customFields?: CustomField[];
```
Comment (samutoljamo-pareto) [inline tools/data-handler/src/interfaces/resource-interfaces.ts:27]: Maybe not part of this PR, but we could consider making these required in typescript and add an empty array if the value is missing. It's much easier to work with empty arrays rather than undefined values

**Review by samutoljamo-pareto — APPROVED** [review]
Looks good to me

File: tools/data-handler/src/interfaces/resource-interfaces.ts (hunk 1 above)
Comment (samutoljamo-pareto) [reply tools/data-handler/src/interfaces/resource-interfaces.ts:27]: Yes

Comment (samutoljamo-pareto) [reply tools/data-handler/src/interfaces/resource-interfaces.ts:27]: https://cyberismo.atlassian.net/jira/software/projects/INTDEV/boards/4?selectedIssue=INTDEV-543
